app/controllers/application_controller.rb

```diff
@@ -1,2 +1,31 @@
 class ApplicationController < ActionController::API
+  before_action :authorize_request
+  attr_reader :decrypted_body
+
+  # rubocop:disable Metrics/MethodLength
+  def authorize_request
+    encrypted_payload = request.raw_post
+    return render_unauthorized if encrypted_payload.nil? || encrypted_payload.empty?
+
+    begin
+      @decrypted_body = JWE.decrypt(encrypted_payload, jwe_key)
+    rescue JWE::DecodeError => e
+      logger.info("retuning unauthorized due to JWE::DecodeError '#{e}'")
+      render_unauthorized
+    rescue JWE::InvalidData => e
+      logger.error("retuning unauthorized due to JWE::InvalidData (we could be missing the decryption key) '#{e}'")
```

Comment:
Spelling of `retuning` here and L13.

Reply:
thanks for the tip will look up shared contexts